src/superstring/core.clj

```diff
@@ -713,3 +713,173 @@
+          \%          "%25"
+          \-          "%2D"
+          \.          "%2E"
+          \<          "%3C"
+          \>          "%3E"
+          \\          "%5C"
+          \^          "%5E"
+          \_          "%5F"
+          \`          "%60"
+          \{          "%7B"
+          \|          "%7C"
+          \}          "%7D"
+          \~          "%7E"
+          \¢          "%A2"
+          \£          "%A3"
+          \¥          "%A5"
+          \¦          "%A6"
+          \§          "%A7"
+          \«          "%AB"
+          \¬          "%AC"
+          \¯          "%AD"
+          \º          "%B0"
+          \±          "%B1"
+          \ª          "%B2"
+          \´          "%B4"
+          \µ          "%B5"
+          \»          "%BB"
+          \¼          "%BC"
+          \½          "%BD"
+          \¿          "%BF"
+          \À          "%C0"
+          \Á          "%C1"
+          \Â          "%C2"
+          \Ã          "%C3"
+          \Ä          "%C4"
+          \Å          "%C5"
+          \Æ          "%C6"
+          \Ç          "%C7"
+          \È          "%C8"
+          \É          "%C9"
+          \Ê          "%CA"
+          \Ë          "%CB"
+          \Ì          "%CC"
+          \Í          "%CD"
+          \Î          "%CE"
+          \Ï          "%CF"
+          \Ð          "%D0"
+          \Ñ          "%D1"
+          \Ò          "%D2"
+          \Ó          "%D3"
+          \Ô          "%D4"
+          \Õ          "%D5"
+          \Ö          "%D6"
+          \Ø          "%D8"
+          \Ù          "%D9"
+          \Ú          "%DA"
+          \Û          "%DB"
+          \Ü          "%DC"
+          \Ý          "%DD"
+          \Þ          "%DE"
+          \ß          "%DF"
+          \à          "%E0"
+          \á          "%E1"
+          \â          "%E2"
+          \ã          "%E3"
+          \ä          "%E4"
+          \å          "%E5"
+          \æ          "%E6"
+          \ç          "%E7"
+          \è          "%E8"
+          \é          "%E9"
+          \ê          "%EA"
+          \ë          "%EB"
+          \ì          "%EC"
+          \í          "%ED"
+          \î          "%EE"
+          \ï          "%EF"
+          \ð          "%F0"
+          \ñ          "%F1"
+          \ò          "%F2"
+          \ó          "%F3"
+          \ô          "%F4"
+          \õ          "%F5"
+          \ö          "%F6"
+          \÷          "%F7"
+          \ø          "%F8"
+          \ù          "%F9"
+          \ú          "%FA"
+          \û          "%FB"
+          \ü          "%FC"
+          \ý          "%FD"
+          \þ          "%FE"
+          \ÿ          "%FF"}))
+
+(defn url-encode
+  "Returns the s string URL encoded. If s is nil, nil is returned.
+  It accepts a :strict? flag (false by default) to encode only URL reserved
+  characters as per https://tools.ietf.org/html/rfc3986#section-2.2."
+  ^String [^String s & {:keys [strict?]}]
```

Comment:
I don't like vararg functions like this because if you give them the wrong arity the linter can't tell.  One way to solve this is to do `:as opts` and then check manually, but I prefer just adding another aritiy with a second argument `strict?`.  This is what I've done for the functions which are optionally case-insensitive.

A good practice then is to prefer:

```clj
(url-encode s :strict)
```

Over:

```cjl
(url-encode s true)
```

To make the call-site more self-documenting.